_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -2465,6 +2468,19 @@ mfxStatus ImplementationAvc::AsyncRoutine(mfxBitstream * bs)
         DdiTaskIter task = FindFrameToStartEncode(m_video, m_lookaheadFinished.begin(), m_lookaheadFinished.end());
         assert(task != m_lookaheadFinished.end());
 
+        mfxU8 numReorderFrames = GetNumReorderFrames(m_video);
+        if (task->m_encOrder < numReorderFrames)
+        {
+            mfxU64 firstTimeStamp = m_decodeTimeStamps.front();
+            mfxU8 distToFirstTimeStamp = numReorderFrames-task->m_encOrder;
```

Comment:
Finally, I'm OK with this implementation :) There are few cosmetic suggestions:
(1) It would be good to explain what happens here in comment.
(2) Also maybe it's good to add intermediate local variable like distInTickUnits = 2 * distToFirstTimeStamp to describe meaning of 2x multiplication.